blueprints/incredible-squaring/src/eigenlayer.rs

```diff
@@ -0,0 +1,472 @@
+#![allow(dead_code)]
+use alloy_contract::{ContractInstance, Interface};
+use alloy_network::Ethereum;
+use alloy_network::EthereumWallet;
+use alloy_primitives::{Address, ChainId, FixedBytes};
+use alloy_primitives::{Bytes, U256};
+use alloy_provider::ProviderBuilder;
+use alloy_provider::Provider;
+use alloy_signer::k256::ecdsa::SigningKey;
+use alloy_signer_local::PrivateKeySigner;
+use alloy_sol_types::private::alloy_json_abi::JsonAbi;
+use alloy_sol_types::sol;
+use color_eyre::{eyre::eyre, eyre::OptionExt, Result};
+use gadget_sdk::job;
+use gadget_sdk::{
+    events_watcher::{
+        evm::{Config, EventWatcher},
+    },
+    keystore::Backend,
+    network::setup::{start_p2p_network, NetworkConfig},
+    run::GadgetRunner,
+};
+use std::convert::Infallible;
+use std::ops::Deref;
+use std::sync::OnceLock;
+use alloy_signer::k256::elliptic_curve::SecretKey;
+use alloy_transport::Transport;
+use IncredibleSquaringTaskManager::{
+    respondToTaskCall, G1Point, G2Point, IncredibleSquaringTaskManagerInstance,
+    NonSignerStakesAndSignature, Task, TaskResponse,
+};
+use gadget_sdk::{
+    tangle_subxt::tangle_testnet_runtime::api::{
+        runtime_types::{sp_core::ecdsa, tangle_primitives::services},
+    },
+    tx,
+};
+use eigensdk_rs::eigen_utils::types::{operator_id_from_key_pair, OperatorPubkeys};
+use eigensdk_rs::eigen_utils::*;
+use eigensdk_rs::incredible_squaring_avs::operator::*;
+use sp_core::Pair;
+use subxt_signer::ecdsa::SecretKeyBytes;
+use gadget_sdk::config::{
+    GadgetConfiguration, StdGadgetConfiguration,
+};
+use gadget_sdk::network::gossip::GossipHandle;
+use gadget_sdk::tangle_subxt::tangle_testnet_runtime::api;
+use gadget_sdk::tangle_subxt::tangle_testnet_runtime::api::runtime_types::tangle_primitives::services::PriceTargets;
+
+// Codegen from ABI file to interact with the contract.
+sol!(
+    #[allow(missing_docs)]
+    #[sol(rpc)]
+    IncredibleSquaringTaskManager,
+    "contracts/out/IncredibleSquaringTaskManager.sol/IncredibleSquaringTaskManager.json"
+);
+
+/// A wrapper around [`IncredibleSquaringTaskManagerInstance`] that can be dereferenced to its [`ContractInstance`].
+#[derive(Debug, Clone)]
+struct IncredibleSquaringInstance<T, P> {
```

Comment:
This seems like it can become auto-generated in the `job` macro no? Since we put these things directly in the macro already it should be.